graphyte.py

```diff
@@ -169,7 +170,12 @@ def _thread_loop(self):
             if current_time - last_check_time >= self.interval:
                 last_check_time = current_time
                 if messages:
-                    self.send_socket(b''.join(messages))
+                    while messages:
+                        batch_messages = []
+                        count_to_send = min(self.send_batchsize,len(messages))
+                        for x in range(count_to_send):
+                            batch_messages.append( messages.pop(0) )
+                        self.send_socket(b''.join(batch_messages))
                     messages = []
 
         # Send any final messages before exiting thread
```

Comment:
I believe we need to duplicate the batching logic down here when sending the final messages as well. I think just duplicating the simpler for loop would be fine, or extracting it to a helper function would be fine too.

Reply:
The volume that could come in while it's not batched should be really small, which is why I hadn't done it, but you're right, it should be avoided to make the contract accurate from the sender api